short-tandem repeat reports

aarthi-mohan opened the pull request:
Summary: Generate family-wise EH reports after bcbio-align step
Steps:
1. Run Expansion Hunter on each sample BAM (output: JSON, VCF, BAM)
2. Combine sample-wise JSON to multi-sample profile 
3. Annotate disease threshold
4. Transpose above TSV, add GT from 1000Genomes EH run (1000G_EH_v1.0.tsv), add sample names in outlier column where sample GT exceeds disease threshold

Usage:
`sh crg.str.sh <familyID>`
The above script will call `crg.eh.sh` which will create a directory structure: `<familyID>/str/expansion_hunter` and final report named `<familyID>_EH_v1.1.xlsx`. Sample report here `/hpf/largeprojects/ccmbio/aarthi/proj_CHEO/CRG/EH/test/908/str/expansion_hunter/908_EH_v1.1.xlsx` 

Note: Disease annotation files are copied to `/hpf/largeprojects/ccmbio/aarthi/proj_CHEO/CRG/str/tandem_repeat_disease_loci_v1.1.tsv` and  double dashes (comes from initial export from Excel to TSV) replaced with hyphen so Python script interprets them correctly.

aarthi-mohan committed:
search chr prefix

aarthi-mohan committed:
commendted out EHDN submission

aarthi-mohan committed:
added precomputed EH GT for 1000G(v1.0 catalog); script for per-family report + fixed bugs

aarthi-mohan committed:
added header line + skipping header line while reading

aarthi-mohan committed:
replaced em_dash with hyphen

Madelinehazel reviewed (approved):
This looks good to me! My only comment was going to be to add the scripts in /hpf/largeprojects/ccmbio/aarthi/proj_CHEO/CRG/str the repo, but it sounds like these were Brett's scripts, so for now this is OK.

aarthi-mohan committed:
moved catalogs and scripts from Brett to crg

aarthi-mohan committed:
moved annotation files to common location; added timestamp to final xlsx filename

aarthi-mohan committed:
moved annotation files to common location; added code blocks to handle request from crg or crg2 pipelines

aarthi-mohan committed:
scripts to perform DBSCAN clustering on EHDN output

aarthi-mohan committed:
scripts from Brett for EHDN

aarthi-mohan committed:
main script to submit jobs for EH & EHDN reports

aarthi-mohan committed:
moved these to common location on hpf

aarthi-mohan commented:
Summary:

- Brett and Bank were okay to upload their scripts here, so moved all those scripts to `crg/str` folder. 
- Moved annotation files to a common location: `/hpf/largeprojects/ccm_dccforge/dccdipg/Common/annotation/ExpansionHunter, /hpf/largeprojects/ccm_dccforge/dccdipg/Common/annotation/ExpansionHunterDenovo`
- Added wrapper bash scripts for EHDN report generation using same script as in EH: `sh crg.str.sh <familyID>`
`~/crg/crg.ehdn.sh`  Runs EHDN v0.7 and creates a TSV file 
`~/crg/ehdn_report.sh` Performs DBSCAN clustering & outlier detection, annotates OMIM, gnoMAD with Annovar and formats the output as `<family>.EHDN.<DATE>.xlsx`
- EH output filename is changed to `<family>.EH-v1.1.<DATE>.xlsx`

aarthi-mohan committed:
log commands

aarthi-mohan committed:
removes rows that contain only control 1000G samples

aarthi-mohan committed:
added 1000G manifest path reqd by format_for_annovar.py

aarthi-mohan committed:
removed duplicate column

aarthi-mohan committed:
added section for STR reports

aarthi-mohan committed:
commented print lines

aarthi-mohan committed:
split outlier and size column sample-wise
